test/beginend-dired-test.el

```diff
@@ -33,80 +33,110 @@
 (require 'dired)
 (require 'dired-x)
 
-(defun beginend-create-dired-buffer ()
-  "Create and return a buffer ressembling a dired one."
-  (assess-with-filesystem '("dir1/" "dir2/")
-    (dired default-directory)
-    (current-buffer)))
-
-(defun beginend-dired-test (&rest spec)
-  "Test dired from SPEC.
-SPEC is a list of symbols representing features to activate."
-  (let* ((begin-p (seq-contains spec 'begin-p))
-         (omit (seq-contains spec 'omit))
-         (hide-details (seq-contains spec 'hide-details))
-         (hide-info-lines (not (seq-contains spec 'keep-info-lines)))
-         (dired-hide-details-hide-information-lines hide-info-lines)
-         (dired-omit-verbose nil))
-    (with-current-buffer (beginend-create-dired-buffer)
-      (goto-char 2) ;; any position
-      (when omit
-        (dired-omit-mode))
-      (when hide-details
-        (dired-hide-details-mode))
-      (if begin-p
-          (progn
-            (beginend-dired-mode-goto-beginning)
-            (expect (looking-at "dir1") :to-be-truthy))
-        (beginend-dired-mode-goto-end)
-        (expect (looking-at "dir2") :to-be-truthy))
-      (kill-buffer))))
+(defun beginend-dired--set-file-times (filename timestamp)
+  "Set times of file FILENAME to TIMESTAMP.
+Return nil on success, else signal an error."
+  (unless (set-file-times filename timestamp)
+    (error "Unable to set directory times: %s" filename)))
+
+(defun beginend-create-dired-buffer (specs entries)
+  "Create and return a buffer ressembling a dired one.
+
+SPECS is a list of symbols which indicates how dired works:
+
+  `dotfiles' means the list contains dotfiles.
+
+ENTRIES specifies the content of the file list.  The order is important.  The
+entries time will be set by that order.  The first one is the newest, and the
+last one is the oldest.  Since the file list is sorted by mtime, the order is
+consistent with the entries order.
+
+Put . and .. in ENTRIES, because the test result depends on their positions."
```

Comment:
please surround . and .. with ticks:
```
Put `.' and `..' in ENTRIES, because the test result depends on their positions."
```

Reply:
I don't understand why you say that even though some tests don't include . and .. in ENTRIES.